analytics/lib/counts.py

```diff
@@ -433,6 +433,33 @@ def pull_function(
     return DataCollector(output_table, pull_function)
 
 
+def count_upload_space_used_by_realm_query(realm: Optional[Realm]) -> QueryFn:
+    if realm is None:
+        realm_clause: Composable = SQL("")
+    else:
+        realm_clause = SQL("zerver_attachment.realm_id = {} AND").format(Literal(realm.id))
+
+    # TODO: What about ArchivedAttachments?
```

Comment:
I'm just thinking whether this leaves there is a meaningful issue in this chain of events:
1. some attachments from today get moved to `ArchivedAttachment` via message deletions 
2. At the end of the day, `update_analytics_counts` runs calculating the space used without the `ArchivedAttachment`s
3. The messages and attachments are restored tomorrow. Now the `RealmCount` record about space used yesterday is incorrect.

But i guess this is pretty irrelevant, since the next calculation of this `RealmCount`  by  `update_analytics_counts` will sum through the entire Attachment table anyway, so the next day's record will be correct anyway. Plus this is a very rare scenario i describe above - and you could have the same concerns for `ArchivedMessage` in all the message-counting countstats.

Reply:
Yeah, this seems correct.